src/stan-users-guide/style-guide.Rmd

```diff
@@ -314,74 +319,98 @@ layout.
 
 ### Line breaks between statements and declarations {-}
 
-It is dispreferred to have multiple statements or declarations on the
-same line, as in the following example.
+Each statement of a program should appear on its own line.
+Declaring multiple variables of the same type can be
+accomplished in a single statement with the syntax
 
 ```
-transformed parameters {
-  real mu_centered;  real sigma;
-  mu = (mu_raw - mean_mu_raw);    sigma = pow(tau,-2);
-}
+real mu, sigma;
 ```
 
-These should be broken into four separate lines.
-
 ### No tabs {-}
 
-Stan programs should not contain tab characters.  They are legal and
-may be used anywhere other whitespace occurs.  Using tabs to layout a
+Stan programs should not contain tab characters. Using tabs to layout a
 program is highly unportable because the number of spaces
 represented by a single tab character varies depending on which
 program is doing the rendering and how it is configured.
 
 ### Two-character indents {-}
 
 Stan has standardized on two space characters of indentation, which is
-the standard convention for C/C++ code.  Another sensible choice is
-four spaces, which is the convention for Java and Python.  Just be
-consistent.
+the standard convention for C/C++ code.
 
 ### Space between `if`, `{` and condition
 
-Use a space after `if`s.  For instance, use `if (x < y) ...`, not
-`if(x < y) ...`.
+Use a space after `if`s.  For instance, use `if (x < y) {...`, not
+`if(x < y){ ...`.
 
-### No space For function calls {-}
+### No space for function calls {-}
 
-There is no space between a function name and the function it applies
-to.  For instance, use `normal(0,1)`, not `normal (0,1)`.
+There should not be space between a function name and the arguments 
+it applies to.  For instance, use `normal(0, 1)`, not `normal (0,1)`.
 
 ### Spaces around operators {-}
 
 There should be spaces around binary operators.  For instance, use
 `y[1] = x`, not `y[1]=x`, use `(x + y) * z` not
 `(x+y)*z`.
 
+Unary operators are written without a space, such as in `-x`, `!y`.
+
+### No spaces in type constraints {-}
+Another exception to the above rule is when the operator is the 
+assignment operator `=` used inside a type constraint, such as
+
+```
+real<lower=0> x;
+```
+
+Here, the preferred style is to omit spaces, except for those following commas as in
```

Comment:
We only want to omit spaces around the `=`.  So rather than

```
real<lower=a*x+b> alpha;
```

I think we'd want

```
real<lower=a * x + b> alpha;
```

I still like it with spaces around the equals sign, myself.

Reply:
Good catch. Your suggestion aligns with how the compiler would format it currently, I just hadn't thought of more complicated expressions in the constraint

Of the three obvious options here (
```stan
real<lower=a*x+b> alpha;
real<lower=a * x + b> alpha;
real<lower = a * x + b> alpha;
```
)
I honestly don't have a super strong preference, but I think two is a nice balance between readable and short, even if it is slightly inconsistent. Then again, it's not like we're gonna be policing this style guide

Reply:
I'm OK with (2) as the official style.  It's not going to come up very often.  When these things get complicated, I tend to write functions for the bounds, which would look OK jammed up, as in 

```stan
real<lower=f(a, x, b)> alpha;
```